tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1349,6 +1349,341 @@ def func_scalar(self, xp, a, scalar):
         a /= scalar
 
 
+_bwise_in_out_dtypes = dtype_utils._permutate_dtype_mapping([
```

Comment:
How about naming `_in_out_dtypes_bitwise` and so on for the consistency?

Reply:
Could you add cases for un-mixed dtypes?

Reply:
https://github.com/chainer/chainer/blob/7ee1103ffcc002aa45749befc2a39f6f99e3345f/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1427

un-mixed dtypes are used here, should we add them here?

Reply:
The case in your comment is for in-place version. Isn't it needed also in `_in_out_dtypes_bitwise`?

Reply:
The line in the comment is used for inplace as well as usual.

https://github.com/chainer/chainer/blob/7ee1103ffcc002aa45749befc2a39f6f99e3345f/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1397

https://github.com/chainer/chainer/blob/7ee1103ffcc002aa45749befc2a39f6f99e3345f/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1427

So while checking broadcastable shapes, we use same `dtype` i.e. un-mixed dtypes for both the input. While we check mixed dtypes, 

https://github.com/chainer/chainer/blob/7ee1103ffcc002aa45749befc2a39f6f99e3345f/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1405

https://github.com/chainer/chainer/blob/7ee1103ffcc002aa45749befc2a39f6f99e3345f/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1434

Reply:
Certainly, the whole `_in_out_dtypes_bitwise` covers all cases for un-mixed and mixed dtypes. However, L1397 is the test for broadcast, and not for same dtypes. L1405 is the test for all dtypes combination, so it seems better to include not only mixed cases but also un-mixed cases.

Reply:
Oh. Sure. On it.

Reply:
Done. I have also added cases for `uint8` and `{int8, int16, int32, int64}`